server/routes/activities/record-attendance/handlers/notAttendedReason.ts

```diff
@@ -1,30 +1,72 @@
 import AttendanceStatus from '../../../../enum/attendanceStatus'
 import AttendanceReason from '../../../../enum/attendanceReason'
 import { convertToTitleCase } from '../../../../utils/utils'
+import { YesNo } from '../../../../@types/activities'
 
-export class NotAttendedReason {
-  @Expose()
-  @Type(() => NotAttendedData)
-  @ReasonEnteredForAllPrisoners({ message: 'Enter an absence reason for all prisoners' })
-  @AbsenceReasonRequired({ message: 'Please enter a reason for the absence' })
-  @PayRequired({ message: 'Please specify whether the prisoner should be paid' })
-  @CaseNoteRequired({ message: 'Please enter a case note' })
-  @IncentiveLevelWarningRequired({
-    message: 'Please specify whether this should be recorded as an incentive level warning',
+const getPrisonerName = (args: ValidationArguments) => (args.object as NotAttendedData)?.prisonerName
+
+export class NotAttendedData {
+  prisonerNumber: string
+
+  prisonerName: string
+
+  @IsEnum(AttendanceReason, { message: args => `Select an absence reason for ${getPrisonerName(args)}` })
+  notAttendedReason: AttendanceReason
+
+  @ValidateIf(o => AttendanceReason.SICK === o.notAttendedReason)
+  @IsEnum(YesNo, { message: (args: ValidationArguments) => `Select if ${getPrisonerName(args)} should be paid` })
+  sickPay?: YesNo
+
+  @ValidateIf(o => AttendanceReason.REST === o.notAttendedReason)
+  @IsEnum(YesNo, { message: (args: ValidationArguments) => `Select if ${getPrisonerName(args)} should be paid` })
+  restPay?: YesNo
+
+  @ValidateIf(o => AttendanceReason.OTHER === o.notAttendedReason)
+  @IsNotEmpty({ message: args => `Enter an absence reason for ${getPrisonerName(args)}` })
+  @MaxLength(100, { message: 'Absence reason must be $constraint1 characters or less' })
+  otherAbsenceReason?: string
+
+  @ValidateIf(o => AttendanceReason.OTHER === o.notAttendedReason)
+  @IsEnum(YesNo, {
+    message: (args: ValidationArguments) =>
+      `Select if this is an acceptable absence for ${getPrisonerName(args)} and they should be paid`,
+  })
+  otherAbsencePay?: YesNo
+
+  @ValidateIf(o => AttendanceReason.SICK === o.notAttendedReason)
+  @IsOptional()
+  @MaxLength(100, { message: 'Additional details must be $constraint1 characters or less' })
+  moreDetail?: string
+
+  @ValidateIf(o => AttendanceReason.REFUSED === o.notAttendedReason)
+  @IsNotEmpty({ message: args => `Enter a case note for ${getPrisonerName(args)}` })
+  @MaxLength(4000, { message: 'Case note must be $constraint1 characters or less' })
+  caseNote?: string
+
+  @ValidateIf(o => AttendanceReason.REFUSED === o.notAttendedReason)
+  @IsEnum(YesNo, {
+    message: (args: ValidationArguments) =>
+      `Select if there should be an incentive level warning for ${getPrisonerName(args)}`,
   })
-  notAttendedData: NotAttendedData
+  incentiveLevelWarningIssued?: YesNo
+}
+
+export class NotAttendedForm {
+  @Type(() => NotAttendedData)
+  @ValidateNested({ each: true })
+  notAttendedData: NotAttendedData[]
```

Comment:
To support showing multiple validation errors I've made this an array which allows `class-validator` to validate against all nested prisoner attendance.

It also means we don't need to use custom validators to validate the request object.